lib/src/ui/map_control_buttons.dart

```diff
@@ -53,6 +61,28 @@ class MapZoomButtons extends StatelessWidget {
               ),
               child: const Icon(Icons.remove),
             ),
+            if (showTrackLocation && !kIsWeb) ...[
+              const SizedBox(height: 8),
+              FloatingActionButton(
+                heroTag: 'MapLibreTrackLocationButton',
+                onPressed: () {
+                  permissionManager ??= PermissionManager();
+
+                  if (permissionManager != null) {
+                    PermissionManager().requestLocationPermissions(
+                      explanation:
+                          'We need your location to show it on the map.',
+                    );
+
+                    if (permissionManager!.locationPermissionsGranted) {
+                      controller.enableLocation();
+                      controller.trackLocation();
+                    }
+                  }
+                },
+                child: const Icon(Icons.gps_fixed),
```

Comment:
What might be the best approach to do this? Currently I haven't found anything in the controller to access this information at runtime

Reply:
Let's have 3 different states:
1. gps not fixed: initial state
2. loading: when the button has been clicked.
3. gps fixed: let's assume the location is shown on the map after `await controller.enableLocation()` is finished for now.

We can check if the app has location permission in initState() and directly use state 3 if so.

What do you think?

Reply:
I agree, I'll make the changes

Reply:
> We can check if the app has location permission in initState() and directly use state 3 if so.

Actually to do this we need to access the controller in the initState, also because I would like to enable localization if the localization permission is granted.
At the moment I have not found a way to access the controller in the initState and then call the enableLocation method (the relationship to the context I guess makes this impossible).
But, if we just want to have the gps icon fixed without the active location in the map, that would not be a problem thanks to the PermissionManager